thinkingmachines/geomancer
# Add tests

Resolves #14 

This requires a test `sqlite` database. It's  around 400 MB so I didn't decide to commit it. My suggestion is that everytime it performs a check, it downloads the `sqlite` file then execute `pytest`

## Things I need second opinion on:
- How I loaded the spatialite extension in `backend.cores.sqlite`. I used an event listener, not sure if that's the best solution.
- My test cases in `tests.spells.distance_to_nearest`. There should be a better way to do that, not quite sure yet.

I am also testing on BigQuery, but that can be triggered off via:

```s
$ pytest -m "not bqtest"
```

## How to run the tests

1. Ensure that you have the test `.sqlite` file in `tests/data/source.sqlite`. I'll upload this in GCS soon
2. Go to root directory of the project then run:

```s
$ pytest
```

"works on my machine..."

![Screenshot from 2019-03-11 21-20-15](https://user-images.githubusercontent.com/12949683/54126892-8b2d2500-4443-11e9-8551-ef2c65b505d5.png)


![coverage](https://user-images.githubusercontent.com/12949683/54127197-350cb180-4444-11e9-93ad-3f73fc600f87.png)


## Next steps (for next sprint)

- Write tests for helper functions in `backend.bq` 
- Refactor, if needed, tests in spell
